modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts

```diff
@@ -0,0 +1,288 @@
+      should.exist(backupKeychain.encryptedPrv);
+    });
+
+    it('should fail to generate TSS key chains', async function () {
+      const nockedBitGoKeychain = await nockBitgoKeychain({
+        coin: coinName,
+        userKeyShare,
+        backupKeyShare,
+        userGpgKey,
+      });
+      const bitgoKeychain = await tssUtils.createBitgoKeychain(userGpgKey, userKeyShare, backupKeyShare);
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
+
+      await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        await MPC.keyShare(2, 2, 3),
+        bitgoKeychain,
+        'passphrase').should.be.rejectedWith('Common keychains do not match');
+      await tssUtils.createUserKeychain(
+        userGpgKey,
+        await MPC.keyShare(1, 2, 3),
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase')
+        .should.be.rejectedWith('Common keychains do not match');
+
+      const share = await MPC.keyShare(1, 2, 3);
+      await tssUtils.createBackupKeychain(
+        userGpgKey,
+        share,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase')
+        .should.be.rejectedWith('Common keychains do not match');
+      await tssUtils.createBackupKeychain(
+        userGpgKey,
+        userKeyShare,
+        await MPC.keyShare(2, 2, 3),
+        bitgoKeychain,
+        'passphrase')
+        .should.be.rejectedWith('Common keychains do not match');
+    });
+  });
+
+  // #region Nock helpers
+  async function nockBitgoKeychain(params: {
+    coin: string,
+    userKeyShare: KeyShare,
+    backupKeyShare: KeyShare,
+    userGpgKey: openpgp.SerializedKeyPair<string>,
+  }): Promise<Keychain> {
+    const bitgoCombined = MPC.keyCombine(bitgoKeyShare.pShare, [params.userKeyShare.nShares[3], params.backupKeyShare.nShares[3]]);
+    const userToBitgoShare = await encryptNShare(
+      bitgoKeyShare,
+      1,
+      userGpgKey.publicKey,
+      userGpgKey.privateKey
+    );
+    const backupToBitgoShare = await encryptNShare(
+      bitgoKeyShare,
+      2,
+      userGpgKey.publicKey,
+      userGpgKey.privateKey
+    );
+
+    const bitgoKeychain: Keychain = {
+      id: '3',
+      pub: '',
+      commonKeychain: bitgoCombined.xShare.y + bitgoCombined.xShare.chaincode,
+      keyShares: [
+        {
+          from: 'bitgo',
+          to: 'user',
+          publicShare: userToBitgoShare.publicShare,
+          privateShare: userToBitgoShare.encryptedPrivateShare,
+        },
+        {
+          from: 'bitgo',
+          to: 'backup',
+          publicShare: backupToBitgoShare.publicShare,
+          privateShare: backupToBitgoShare.encryptedPrivateShare,
+        },
+      ],
+    };
+
+    nock(bgUrl)
+      .post(`/api/v2/${params.coin}/key`, _.matches({ keyType: 'tss', source: 'bitgo' }))
+      .reply(200, bitgoKeychain);
+
+    return bitgoKeychain;
+  }
+
+  async function nockUserKeychain(params: {
```

Comment:
the nockUserKeychain and backupkeychain seems to have similar logic, then lets combine them to something like nockKeychain and pass the difference as parameters